### ISSUE TITLE
GPT2 attention mask and output masking

### ISSUE THREAD

**Issue by anandhperumal** [huggingface/transformers#1573]
## ❓ Questions & Help

I have couple of questions:
1. In the original gpt2 they didn't pad the sequence, so they didn't need a  attention mask, but in other cases where we our input sequence is small and we pad the input, don't we need a attention mask?
2. I have padded the labels in the left with -1, in the cost function how do I skip the padded elements in labels? and same for logits how do I skip the padded elements ?

**Comment by anandhperumal** [comment]
Closing this because I found my answer

**Comment by sli613** [comment]
Hi, would you mind sharing what the answer you found is? Thank you so much!

**Comment by anandhperumal** [comment]
Sorry for the delay. Gpt2 was trained as a CLM model with a fixed block size of data. So there was no need for attention mask. (That is what I understood).
